docs/AUTH_API.md

```diff
@@ -0,0 +1,73 @@
+# Authentication API
+
+For now this is just a dumping ground of ideas how to take the [advanced JSON login protocol](AUTH_PROTOCOL.md) and turn it into an API to be used by the rest of the system. Eventually I expect that this would morph into a doc page that more conherently explains the components I propose here and explains how they all work together to provide a new `systemd-authd` (name TBD?) component.
+
+## `io.systemd.AuthenticationProvider`
+
+This Varlink interface has a single method: `Authenticate(username | uid)`. It returns multiple values:
+
+1. First it sends over the file descriptor. This can be sent off to the frontend
+2. Later, once the frontend has finished authentication, it returns either a `AuthenticationFailed` error or success. In the success case, it returns a blob of data to be used for things like home directory encryption/decryption and whatnot.
+
+The providers are dropped into `/run/systemd/auth-provider/[name]`. You can decide which provider should be used by looking at the `authService` field of the user's user record.
+
+## `/usr/lib/systemd/systemd-authd-pam-bridge`
+
+For various display managers to port to this new mechanism, they need to be able to maintain compatibility with users that don't have an `AuthenticationProvider` associated and potentially some custom PAM authentication plugin (i.e. imagine someone who previously implemented two-factor by having `pam_unix.so` followed by `pam_google_authenticator.so`). To that end, we need to have backwards-compatibility for running things like this.
+
+This is what `systemd-authd-pam-bridge` is for. If we don't have an `AuthenticationProvider` associated with a given user, then each service (GDM, Plasma DM/SDDM, LightDM, etc) can fall back to PAM.
```

Comment:
Here I kinda see some room to do something that I was thinking on for a while, that could also improve the security of those systems standard PAM systems, by allow to define the resource control for such process, depending on the pam service. And basically sandboxing the PAM service application binary as much as we can.

Reply:
Yeah for sure. In my mind this wouldn't even have a systemd service defined, and would instead it would be up to GDM/SDDM/etc to use the systemd-provided binary to define its own service